### PR TITLE
UPGRADING RAILS TO 4.2.10

### PR DESCRIPTION
**Pull request by Reinkaos**
Upgrading the version of Rails that the Manager is using from 3.2.22 to 4.2.10.

### CODE REVIEW

File: app/models/abstract_job.rb [hunk 1]
```diff
@@ -10,7 +10,8 @@ class AbstractJob < ActiveResource::Base
   include EnvironmentHelpers
   
   self.site = ENV["WORKER_HOST"]
-  self.user = ENV["WORKER_API_KEY"]
+  headers['Authorization'] = "Token token=#{ENV['WORKER_API_KEY']}"
```
Comment (Reinkaos) [inline app/models/abstract_job.rb]: This will also need to be part of the authorization story.

File: app/models/version.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 # http://digitalnz.org/supplejack
 
 class Version
-
   include Mongoid::Document
   include Mongoid::Timestamps::Created
   include Mongoid::Paranoia
-  include ActiveModel::SerializerSupport
+  include Mongoid::Attributes::Dynamic
+  # include ActiveModel::SerializerSupport
```
Comment (Reinkaos) [inline app/models/version.rb]: Remove me.

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -24,6 +24,10 @@
   # require 'devise/orm/active_record'
   require 'devise/orm/mongoid'
 
+  config.secret_key = '9c3973a3344d174711d9f434f55d82e99322c23b8faf56b97b30bec9d3fc3bd34bfd01119c981d96daba24e63e69ad2f28785baad403caf613d671891523e745'
```
Comment (Reinkaos) [inline config/initializers/devise.rb]: Remove top one.

File: spec/controllers/snippets_controller_spec.rb [hunk 4]
```diff
@@ -1,136 +1,138 @@
...
-      Snippet.stub(:find_by_name) { snippet }
-    end
-
-    it "should find the current version of the snippet" do
-      Snippet.should_receive(:find_by_name).with("Copyright", :staging) { snippet }
-      get :current_version, name: "Copyright", environment: :staging, format: :js
-      assigns(:snippet).should eq snippet
-    end
-  end
+  # describe "GET 'new'" do
```
Comment (Reinkaos) [inline spec/controllers/snippets_controller_spec.rb]: Check why these are commented out.

File: spec/support/devise.rb [hunk 5]
```diff
@@ -1,11 +1,22 @@
...
+# See https://github.com/DigitalNZ/supplejack_manager for details.
+#
+# Supplejack was created by DigitalNZ at the National Library of NZ and the Department of Internal Affairs.
 # http://digitalnz.org/supplejack
 
 RSpec.configure do |config|
-  config.include Devise::TestHelpers, :type => :controller
+  config.include Devise::Test::ControllerHelpers, type: :controller
+
+  # rspec-rails 3 will no longer automatically infer an example group's spec type
```
Comment (Reinkaos) [inline spec/support/devise.rb]: Remove commented out stuff.

File: app/controllers/parser_versions_controller.rb [hunk 6]
```diff
@@ -22,13 +25,13 @@ def current
   end
 
   def show
-    @harvest_job = HarvestJob.build(parser_id: @parser.id, version_id: @version.id, user_id: current_user.id)
-    @enrichment_job = EnrichmentJob.build(parser_id: @parser.id, version_id: @version.id, user_id: current_user.id)
+    @harvest_job = HarvestJob.build(parser_id: @parser.id, version_id: @version.id)
```
[the remark this answers is not in the document]
Comment (Reinkaos) [reply app/controllers/parser_versions_controller.rb]: Seems to be ok!

**Comment by motizuki** [comment]
👍

**Pull request merged**